src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+  styleUrl: "gauge.scss",
+  tag: "gx-gauge"
+})
+export class Gauge implements IComponent {
+  @Element() element: HTMLElement;
+
+  @Event() gxGaugeDidLoad: EventEmitter;
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * This property allows you to select the gauge type. _(Circle or Line)_.
+   * Default is linear type.
+   */
+  @Prop() gaugeType: "line" | "circle" = "line";
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  @State() maxValue: number;
+
+  @State() totValues = this.minValue;
+
+  @State() children = [];
+
+  @State() minorSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.children = [...this.children, childRange];
+    this.totValues += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1);
+      this.totValues -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1, childRange);
+      this.totValues = 0;
+      for (const childInstance of this.children) {
+        this.totValues += childInstance.amount;
+      }
+    });
+  }
+
+  componentWillLoad() {
+    this.totValues = 0;
+  }
+
+  private calcThickness() {
+    return typeof this.thickness === "number" &&
+      (this.thickness > 0 && this.thickness <= 100)
+      ? this.thickness / 5
+      : 10;
+  }
+
+  private calcPercentage() {
+    return (
+      (this.currentValue - this.minValue) *
+      100 /
+      (this.maxValue - this.minValue)
+    );
+  }
+
+  private renderCircle(childRanges) {
+    const svgRanges = [];
+    let acumulation = 0;
+    //////////////////////////////////////////
+    function calcPositionRange(preValue, value) {
+      acumulation += preValue;
+      return value + acumulation;
+    }
+
+    function addSVGCircle(currentChild, nextChild, component) {
+      return (
+        <circle
+          r="39.59%"
+          cx="50%"
+          cy="50%"
+          stroke={currentChild.color}
+          stroke-dasharray={`${2.488 /
+            (component.totValues / 100) *
+            calcPositionRange(
+              !!nextChild ? parseInt(nextChild.getAttribute("amount"), 10) : 0,
```

Comment:
If you are checking if nextChild is a truthy value, you don't need to double negate it.

Reply:
I make a double negate to obtain true or false according to if exist or not the nextChild.

`
if nextChild exist => true => true option of ternary
`
`
if nextChild is undefined => false => false option of ternary
`

Reply:
I understand, but what I'm suggesting is that you can achieve the same result without the double negation.

Reply:
This I fix in the 4700ea25a372abf35b3dfc54b371e0188315facd commit